.circleci/config.yml

```diff
@@ -2,42 +2,80 @@
 
 version: 2.1
 
-steps: &steps
+.install_bundler: &install_bundler
+  run:
+    name: Install a specific bundler version
+    command: gem install bundler -v 1.16.6
+
+.copy_current_gemfile: &copy_current_gemfile
+  run:
+    name: Cat current Gemfile to a specific file so caching works
+    command: cat "$BUNDLE_GEMFILE.lock" > current_gemfile.lock
+
+.restore_cache: &restore_cache
+  restore_cache:
+    keys:
+      - bundler-dependencies-{{ checksum "current_gemfile.lock" }}-{{ .Environment.CIRCLE_JOB }}
+
+.install_dependencies: &install_dependencies
+  run:
+    name: Install dependencies
+    command: bundle install --without development --path /home/circleci/project/vendor/bundle --retry 3 --jobs 3
+
+.install_chromedriver: &install_chromedriver
+  run:
+    name: Install chromedriver
+    command: bin/install_chromedriver.sh
+
+.save_cache: &save_cache
+  save_cache:
+    key: bundler-dependencies-{{ checksum "current_gemfile.lock" }}-{{ .Environment.CIRCLE_JOB }}
+    paths:
+      - vendor/bundle
+
+.run_tests: &run_tests
+  run:
+    name: Run tests
+    command: COVERAGE=true bundle exec rake
+
+.generate_docs: &generate_docs
+  run:
+    name: Generate docs
+    command: bundle exec rake docs:build
+
+.run_lints: &run_lints
+  run:
+    name: Run lints
+    command: bundle exec rake lint
+
+.test_steps: &test_steps
   - checkout
+  - *install_bundler
+  - *copy_current_gemfile
+  - *restore_cache
+  - *install_dependencies
+  - *install_chromedriver
+  - *save_cache
+  - *run_tests
 
-  - run:
-      name: Install a specific bundler version
-      command: gem install bundler -v 1.16.6
-
-  - run:
-      name: Cat current Gemfile to a specific file so caching works
-      command: cat "$BUNDLE_GEMFILE.lock" > current_gemfile.lock
-
-  - restore_cache:
-      keys:
-        - bundler-dependencies-{{ checksum "current_gemfile.lock" }}-{{ .Environment.CIRCLE_JOB }}
-
-  - run:
-      name: Install dependencies
-      command: bundle install --without development --path /home/circleci/project/vendor/bundle --retry 3 --jobs 3
-
-  - run:
-      name: Install chromedriver
-      command: bin/install_chromedriver.sh
+jobs:
+  lint_and_docs:
+    docker:
+      - image: circleci/ruby:2.5.1
```

Comment:
I noticed here and in other places too we are using Ruby 2.5.1 but 2.5.3 has already been released. Should we open a new PR to update that? Or do you prefer we keep at 2.5.1 for the time being?

Reply:
Yeah, I almost used 2.5.3 here, but decided to stay with 2.5.1 for now, and separately update everything on a separate PR.

Reply:
It occurred to me that we weren't on Ruby 2.5.2 but saw now that it was released only days ago and that 2.5.3 was to correct an issue in that patch release. I was just wondering why we hadn't updated sooner but now it makes sense. Sounds good!